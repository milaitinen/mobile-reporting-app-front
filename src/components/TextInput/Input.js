import React from 'react';
import { View, TextInput } from 'react-native';
import Icon from 'react-native-vector-icons/Feather';

import styles from './styles';

const Input = (props) => {
    return (
        <View
            style={styles.InputContainer}>
            <Icon name={props.name} color={'white'} size={16} />
            <TextInput
                autoCapitalize = 'none'
                onChangeText={props.onChangeText}
                underlineColorAndroid='transparent'
                style={styles.InputStyle}
                placeholderTextColor='white'
                {...props}
            />
        </View>
    );
};

export default Input;