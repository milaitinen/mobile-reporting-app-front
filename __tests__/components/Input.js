import 'react-native';
import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import sinon from 'sinon';


import Input from '../../src/components/TextInput/Input';

configure({ adapter: new Adapter() });

describe('Input component', () => {
    it('should render correctly when not focused', () => {
        const tree = renderer.create(
            <Input
                placeholder={'Testi'}
                placeholderTextColor={'gray'}
                value={null}
                onChangeText={jest.fn()}
            />
        );
        expect(tree).toMatchSnapshot();
    });

    it('should render correctly when focused', () => {
        const tree = renderer.create(
            <Input
                placeholder={'Testi'}
                placeholderTextColor={'gray'}
                value={null}
                onChangeText={jest.fn()}
            />
        );
        tree.getInstance().handleFocus();
        expect(tree).toMatchSnapshot();
    });

    it('should render correctly when multiline', () => {
        const tree = renderer.create(
            <Input
                multiline
                placeholder={'Testi'}
                placeholderTextColor={'gray'}
                value={null}
                onChangeText={jest.fn()}
            />
        );
        expect(tree).toMatchSnapshot();
    });

    it('should handle focus', () => {
        const spy = sinon.spy(Input.prototype, 'handleFocus');
        const input = shallow(
            <Input
                placeholder={'Testi'}
                placeholderTextColor={'gray'}
                value={null}
                onChangeText={jest.fn()}
            />)
        ;
        input.find('TextInput').simulate('focus');
        expect(spy.called).toEqual(true);
    });

    it('should handle blur', () => {
        const spy = sinon.spy(Input.prototype, 'handleBlur');
        const input = shallow(
            <Input
                placeholder={'Testi'}
                placeholderTextColor={'gray'}
                value={null}
                onChangeText={jest.fn()}
            />)
        ;
        input.find('TextInput').simulate('blur');
        expect(spy.called).toEqual(true);
    });
});