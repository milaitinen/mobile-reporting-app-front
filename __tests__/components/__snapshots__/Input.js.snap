// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component should render correctly when focused 1`] = `
<TextInput
  allowFontScaling={true}
  onBlur={[Function]}
  onChangeText={[Function]}
  onFocus={[Function]}
  placeholder="Testi"
  placeholderTextColor="gray"
  selectionColor={undefined}
  style={
    Array [
      undefined,
      undefined,
    ]
  }
  underlineColorAndroid="transparent"
  value={null}
/>
`;

exports[`Input component should render correctly when multiline 1`] = `
<TextInput
  allowFontScaling={true}
  multiline={true}
  onBlur={[Function]}
  onChangeText={[Function]}
  onFocus={[Function]}
  placeholder="Testi"
  placeholderTextColor="gray"
  selectionColor={undefined}
  style={undefined}
  underlineColorAndroid="transparent"
  value={null}
/>
`;

exports[`Input component should render correctly when not focused 1`] = `
<TextInput
  allowFontScaling={true}
  onBlur={[Function]}
  onChangeText={[Function]}
  onFocus={[Function]}
  placeholder="Testi"
  placeholderTextColor="gray"
  selectionColor={undefined}
  style={undefined}
  underlineColorAndroid="transparent"
  value={null}
/>
`;
